src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java

```diff
@@ -335,6 +336,19 @@ public SkyValue compute(SkyKey key, Environment env)
                 getPrioritizedDetailedExitCode(causes)));
       }
 
+      RuleConstraintSemantics.IncompatibleTargetCreationResult incompatibleTargetCreationResult =
+          RuleConstraintSemantics.createDirectlyIncompatibleTarget(
+              targetAndConfiguration,
+              configConditions,
+              env,
+              unloadedToolchainContexts,
+              state.transitivePackagesForPackageRootResolution);
+      if (incompatibleTargetCreationResult.needsMoreConfiguredTargets()) {
```

Comment:
(This leads from an earlier comment) Instead of having needsMoreConfiguredTargets, createDirectlyIncompatibleTarget should just return null to signal the need for a skyframe restart.

Reply:
I would like to simplify it down to that if possible. Let's discuss further in the earlier comment.